include/crocoddyl_msgs/conversions.h

```diff
@@ -656,9 +656,93 @@ fromMsg(const pinocchio::ModelTpl<double, Options, JointCollectionTpl> &model,
+ * @param locked_joint_ids[in]  Ids of the locked joints
+ */
+template <int Options, template <typename, int> class JointCollectionTpl>
+static inline void fromReduced(
+    const pinocchio::ModelTpl<double, Options, JointCollectionTpl> &model,
+    const pinocchio::ModelTpl<double, Options, JointCollectionTpl>
+        &reduced_model,
+    Eigen::Ref<Eigen::VectorXd> q_out, Eigen::Ref<Eigen::VectorXd> v_out,
+    Eigen::Ref<Eigen::VectorXd> tau_out,
+    const Eigen::Ref<const Eigen::VectorXd> &q_in,
+    const Eigen::Ref<const Eigen::VectorXd> &v_in,
+        const Eigen::Ref<const Eigen::VectorXd> &tau_in,
+    const Eigen::Ref<const Eigen::VectorXd> &qref,
+    const std::vector<pinocchio::JointIndex> &locked_joint_ids) {
+  const std::size_t root_joint_id = getRootJointId(model);
+  const std::size_t nq_root = model.joints[root_joint_id].nq();
+  const std::size_t nv_root = model.joints[root_joint_id].nv();
+  if (q_out.size() != model.nq) {
+    throw std::invalid_argument("Expected q_out to be " +
+                                std::to_string(model.nq) + " but received " +
+                                std::to_string(q_out.size()));
+  }
+  if (q_in.size() != reduced_model.nq) {
+    throw std::invalid_argument("Expected q_in to be " +
+                                std::to_string(reduced_model.nq) +
+                                " but received " + std::to_string(q_in.size()));
+  }
+  if (v_out.size() != model.nv) {
+    throw std::invalid_argument("Expected v_out to be " +
+                                std::to_string(model.nv) + " but received " +
+                                std::to_string(v_out.size()));
+  }
+  if (v_in.size() != reduced_model.nv) {
+    throw std::invalid_argument("Expected v_in to be " +
+                                std::to_string(reduced_model.nv) +
+                                " but received " + std::to_string(v_in.size()));
+  }
+    if (static_cast<std::size_t>(tau_out.size()) != model.nv - nv_root) {
+    throw std::invalid_argument(
+        "Expected tau_out to be " + std::to_string(model.nv - nv_root) +
+        " but received " + std::to_string(tau_out.size()));
+  }
+  if (static_cast<std::size_t>(tau_in.size()) != reduced_model.nv - nv_root) {
+    throw std::invalid_argument(
+        "Expected tau_in to be " + std::to_string(reduced_model.nv - nv_root) +
+        " but received " + std::to_string(tau_in.size()));
+  }
+
+  typedef pinocchio::ModelTpl<double, Options, JointCollectionTpl> Model;
+  typedef typename Model::JointModel JointModel;
+
+  q_out.head(nq_root) = q_in.head(nq_root);
+  v_out.head(nv_root) = v_in.head(nv_root);
+    for (std::size_t j = root_joint_id + 1;
+       j < static_cast<std::size_t>(reduced_model.njoints); ++j) {
+    const std::string &name = reduced_model.names[j];
+    JointModel joint = model.joints[model.getJointId(name)];
+    JointModel reduced_joint = model.joints[reduced_model.getJointId(name)];
```

Comment:
I know this comes from my first implementation; however, it would be convenient (if possible) to do:
```suggestion
    const JointModel& joint = model.joints[model.getJointId(name)];
    const JointModel& reduced_joint = model.joints[reduced_model.getJointId(
```

If this is possible, then please propagate to the other functions.